### PR TITLE
Fixing CMake build

### PR DESCRIPTION
**Pull request by borisfom**
Added 3rdparty/ to include directories list for CMake (used to be only in Makefile)

### CODE REVIEW

**Comment by lukeyeager** [comment]
Addresses the issue reported here - https://github.com/NVIDIA/caffe/pull/62#issuecomment-153503647

File: scripts/travis/travis_build_and_test.sh [hunk 1]
```diff
@@ -6,7 +6,7 @@ set -e
 MAKE="make --jobs=$NUM_THREADS --keep-going"
 
 if $WITH_CMAKE; then
-  mkdir build
+  mkdir -p build
```
Comment (lukeyeager) [inline scripts/travis/travis_build_and_test.sh]: Is this so you can run the script locally? I don't think we need to check that into GitHub.

**Comment by lukeyeager** [comment]
@borisfom You can rebase on `caffe-0.14` to include the TravisCI CMake fix at https://github.com/NVIDIA/caffe/pull/66.

**Comment by lukeyeager** [comment]
AFAICT, this fixes the issue. I'm no longer getting this CMake build error:

    caffe/src/caffe/util/gpu_memory.cpp:14:38: fatal error: cub/cub/util_allocator.cuh: No such file or directory

File: scripts/travis/travis_build_and_test.sh (hunk 1 above)
Comment (borisfom) [inline scripts/travis/travis_build_and_test.sh]: Yes that prevents it from failng when build directory exists - why shouldn't we check it in, it won't break anything.

**Pull request merged**